sherpa/data.py

```diff
@@ -464,17 +487,13 @@ class Data(NoNewAttributesAfterInit, BaseData):
 
     def __init__(self, name, indep, y, staterror=None, syserror=None):
         """
```

Comment:
I think we probably want some description added to the `Data` docstring to describe the handling of masked arrays.